examples/pwa/pwa.html

```diff
@@ -11,6 +11,33 @@
     if (window.location.pathname.indexOf('/examples') == 0) {
       window.location.assign('/pwa');
     }
+
+    function loadScript(url, callback) {
+        // Adding the script tag to the head as suggested before
+        var head = document.getElementsByTagName('head')[0];
+        var script = document.createElement('script');
+        script.type = 'text/javascript';
+        script.src = url;
+
+        // Then bind the event to the callback function.
+        // There are several events for cross browser compatibility.
+        script.onreadystatechange = callback;
+        script.onload = callback;
+
+        // Fire the loading
+        head.appendChild(script);
+    }
+
+    // Using this explicitly, as firefox errors due to our polyfills
+    loadScript("https://cdnjs.cloudflare.com/ajax/libs/webcomponentsjs/1.0.3/webcomponents-sd-ce.js", function() {
+      console.log('Started loading webcomponents polyfill!');
+    });
+
+    setTimeout(function() {
```

Comment:
The trick is: we need to load both the polyfill and pwa.js ASAP and in parallel. But, inside pwa.js, we need to wait until polyfill is ready. But only wait for the narrowest possible scope. E.g. pwa.js may need to do a dozen of initialization steps, but only one of these steps needs polyfill and thus only this step must wait. This is very important since this code explains exactly that - where we have the dependency. And we will need to repeat in similar way for the documentation.